mlrun/package/packagers_manager.py

```diff
@@ -0,0 +1,698 @@
+)
+from mlrun.package.packager import Packager
+from mlrun.package.packagers.default_packager import DefaultPackager
+from mlrun.utils import StorePrefix, logger
+
+from .common import LogHintKey, TypingUtils
+
+
+class PackagersManager:
+    """
+    A packager manager is holding the project's packagers and sending them objects to pack and data items to unpack.
+
+    It prepares the instructions / log hint configurations and then looks for the first packager who fits the task.
+    That's why when the manager collects its packagers, it first collects builtin MLRun packagers and only then the
+    user's custom packagers, this way user's custom packagers will have higher priority.
+    """
+
+    # Mandatory packagers to be collected at initialization time:
+    _MLRUN_REQUIREMENTS_PACKAGERS = [
+        "python_standard_library",
+        "pandas",
+        "numpy",
+        "mlrun",
+    ]
+    # Optional packagers to be collected at initialization time:
+    _EXTENDED_PACKAGERS = ["matplotlib", "plotly", "bokeh"]
+
+    def __init__(self, default_packager: Type[Packager] = None):
+        """
+        Initialize a packagers manager.
+
+        :param default_packager: The default packager should be a packager that fits to all types. It will be the first
+                                 packager in the manager's packagers (meaning it will be used at lowest priority) and it
+                                 should be found fitting when all packagers managed by the manager do not fit an
+                                 object or data item. Default to ``mlrun.DefaultPackager``.
+        """
+        # Set the default packager:
+        self._default_packager = default_packager or DefaultPackager
+
+        # Initialize the packagers list (with the default packager in it):
+        self._packagers: List[Type[Packager]] = [self._default_packager]
+
+        # Set an artifacts list and results dictionary to collect all packed objects (will be used later to write extra
+        # data if noted by the user using the log hint key "extra_data")
+        self._artifacts: List[Artifact] = []
+        self._results = {}
+
+        # Collect the builtin standard packagers:
+        self._collect_packagers()
+
+    @property
+    def artifacts(self) -> List[Artifact]:
+        """
+        Get the artifacts that were packed by the manager.
+
+        :return: A list of artifacts.
+        """
+        return self._artifacts
+
+    @property
+    def results(self) -> dict:
+        """
+        Get the results that were packed by the manager.
+
+        :return: A results dictionary.
+        """
+        return self._results
+
+    def collect_packagers(self, packagers: List[Union[Type, str]]):
+        """
+        Collect the provided packagers. Packagers passed as module paths will be imported and validated to be of type
+        `Packager`. If needed to import all packagers from a module, use the module path with a "*" at the end.
+
```

Comment:
might be worth stating what will be order the packagers in `*` will be added in

Reply:
The order they are collected is being printed in [debug]. I think for now this is enough. If user would like to specify the order, he can use the specific module path and not the `*`.

Reply:
but the order will be in the order that they are defined in the module level right?

Reply:
yes